### PR TITLE
Added Many to Many relation

### PR DESCRIPTION
**Pull request by Daniel-olaO**
Redesigned the schema and remove the repository folder

### CODE REVIEW

File: src/controllers/course.controls.js [hunk 1]
```diff
@@ -9,46 +9,56 @@ const CourseRepository = require('../repositories/courses.repository');
  * @returns {Object} - Success response in JSON
  */
 
-const courseRepository = new CourseRepository()
+// const courseRepository = new CourseRepository()
```
Comment (Russell-Pollari) [inline src/controllers/course.controls.js]:
```suggestion

```

File: src/controllers/student.controls.js [hunk 2]
```diff
@@ -9,68 +9,84 @@ const CourseRepository = require('../repositories/courses.repository');
...
-            const result = await studentRepository.dropCouse(req.params.id, req.params.code);
-            res.status(202).json(result);
-        }
-        catch(error){
-            res.status(400).json(error);
-        }
+  addStudent: async (req, res, next)=>{
+    const id = generateId();
+    const newStudent = new Model({
+      studentId: id,
```
Comment (Russell-Pollari) [inline src/controllers/student.controls.js:17]: question. Every document will be created with a unique `_id`. Is `studentId` a needed?
You will already have a unique id for each student with `student._id`

File: src/controllers/user.controls.js [hunk 3]
```diff
@@ -12,61 +13,76 @@ const { generateAccessToken } = require('../utils/utils');
  * @returns {Object} - Response object
  */
 
-//mddleware for JWT
+// mddleware for JWT
 
 
-
-const uerRepository = new UserRepository(); 
+// const uerRepository = new UserRepository();
```
Comment (Russell-Pollari) [inline src/controllers/user.controls.js]:
```suggestion
```

File: src/database/index.js [hunk 4]
```diff
@@ -1,10 +1,10 @@
 const mongoose = require('mongoose');
-const mongoString = process.env.DB_CONNECTION_URL
+const mongoString = process.env.DB_CONNECTION_URL;
 
 mongoose.connect(mongoString);
 const database = mongoose.connection;
 
-database.on('error', (error) => console.log("error"));
+database.on('error', (error) => console.log('error'));
```
Comment (Russell-Pollari) [inline src/database/index.js]: I think you want to log the `error` object not a string with `'error'`

File: src/database/models/courses.model.js [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
-    program: String,
+  code: String,
+  name: String,
+  professor: String,
+  program: String,
+  students: [
+    {
+      type: mongoose.Types.ObjectId,
+      ref: 'Student',
+    },
```
Comment (Russell-Pollari) [inline src/database/models/courses.model.js]: :+1:

File: src/database/models/students.model.js [hunk 6]
```diff
@@ -1,25 +1,23 @@
...
+  firstName: String,
+  lastName: String,
+  email: String,
+  phone: String,
+  program: String,
+  courses: [
+    {
+      type: mongoose.Types.ObjectId,
+      ref: 'Course',
     },
```
Comment (Russell-Pollari) [inline src/database/models/students.model.js:19]: You want a single source of truth. Either keep this, or keep the `students` field in the `courses` document. But not both

File: src/validations/student.js [hunk 7]
```diff
@@ -1,29 +1,27 @@
...
-    console.log(message)
+exports.validateStudent = function(req, res, next) {
+  const schema = joi.object({
+    firstName: joi.string().required(),
+    lastName: joi.string().required(),
+    email: joi.string()
+        .email().required(),
+    phone: joi.string().length(10).pattern(/^[0-9]+$/).required(),
+    program: joi.string().length(3).required(),
+    // startDate: joi.date().required()
```
Comment (Russell-Pollari) [inline src/validations/student.js]:
```suggestion
```

File: src/controllers/student.controls.js [hunk 8]
```diff
@@ -9,68 +9,84 @@ const CourseRepository = require('../repositories/courses.repository');
...
+      res.status(204).send(deleteStudent);
+    } catch (error) {
+      res.status(400).json(error);
+    }
+  },
+  takeCouse: async (req, res, next)=>{
+    try {
+      const student = await Model.findOne({studentId: req.params.id});
+      const course = await Model.findOne({code: req.params.code});
+      student.courses.push(course);
```
Comment (Russell-Pollari) [inline src/controllers/student.controls.js]: I think you just want to push the `course._id` with the new schema?

**Review by Russell-Pollari — CHANGES_REQUESTED** [review]

File: src/controllers/student.controls.js (hunk 2 above)
Comment (Daniel-olaO) [reply src/controllers/student.controls.js:17]: I want an Id that is also easy to remember

Comment (Russell-Pollari) [reply src/controllers/student.controls.js:17]: HM, okay to leave as-is. But I do think this is a poor design. You don't think `student._id` is easy to remember? Every single mongo document has a unique `_id` by default. And most developers will know this and assume that when you say `studentId` you mean the `_id` of the `student` document

**Review by Russell-Pollari — APPROVED** [review]

**Comment by Russell-Pollari** [comment]
remember to close the relevant issue(s) once merged. You can do this automatically by including in the PR description: `closes #20 `

**Pull request merged**